<?xml version="1.0" ?>
<robot name="go2">
  <link name="base_link">
      </link>
  <joint name="floating_base" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="trunk"/>
  </joint>
  <link name="trunk">
        <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.3762 0.0935 0.114"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.021112 0.0 -0.005366"/>
      <mass value="6.921"/>
      <inertia ixx="0.02448" ixy="0.00012166" ixz="0.0014849" iyy="0.098077" iyz="-3.12e-05" izz="0.107"/>
    </inertial>
  </link>
  <joint name="imu_joint" type="fixed">
    <parent link="trunk"/>
    <child link="imu_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="imu_link">
    <inertial>
      <mass value="0.001"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001"/>
    </inertial>
        <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size=".001 .001 .001"/>
      </geometry>
    </collision>
  </link>
  <joint name="rf_hip_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.1934 -0.0465 0"/>
    <parent link="trunk"/>
    <child link="rf_hip_link"/>
    <axis xyz="1 0 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.0472" upper="1.0472" velocity="30.1"/>
  </joint>
  <link name="rf_hip_link">
        <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 -0.0955 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.046"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0054 -0.00194 -0.000105"/>
      <mass value="0.678"/>
      <inertia ixx="0.00048" ixy="3.01e-06" ixz="1.11e-06" iyy="0.000884" iyz="1.42e-06" izz="0.000596"/>
    </inertial>
  </link>
  <joint name="rf_upper_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.0955 0"/>
    <parent link="rf_hip_link"/>
    <child link="rf_upper_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.5708" upper="3.4907" velocity="30.1"/>
  </joint>
  <link name="rf_upper_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.0245 0.034"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00374 0.0223 -0.0327"/>
      <mass value="1.152"/>
      <inertia ixx="0.00584" ixy="-8.72e-05" ixz="-0.000289" iyy="0.0058" iyz="-0.000808" izz="0.00103"/>
    </inertial>
  </link>
  <joint name="rf_lower_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="rf_upper_leg_link"/>
    <child link="rf_lower_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="35.55" lower="-2.7227" upper="-0.83776" velocity="20.06"/>
  </joint>
  <link name="rf_lower_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.016 0.016"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.00548 -0.000975 -0.115"/>
      <mass value="0.154"/>
      <inertia ixx="0.00108" ixy="3.4e-07" ixz="1.72e-05" iyy="0.0011" iyz="8.28e-06" izz="3.29e-05"/>
    </inertial>
  </link>
  <joint name="rf_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="rf_lower_leg_link"/>
    <child link="rf_foot_link"/>
  </joint>
  <link name="rf_foot_link">
        <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.06"/>
      <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
    </inertial>
  </link>
    <joint name="lf_hip_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.1934 0.0465 0"/>
    <parent link="trunk"/>
    <child link="lf_hip_link"/>
    <axis xyz="1 0 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.0472" upper="1.0472" velocity="30.1"/>
  </joint>
  <link name="lf_hip_link">
        <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0.0955 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.046"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0054 0.00194 -0.000105"/>
      <mass value="0.678"/>
      <inertia ixx="0.00048" ixy="-3.01e-06" ixz="1.11e-06" iyy="0.000884" iyz="-1.42e-06" izz="0.000596"/>
    </inertial>
  </link>
  <joint name="lf_upper_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.0955 0"/>
    <parent link="lf_hip_link"/>
    <child link="lf_upper_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.5708" upper="3.4907" velocity="30.1"/>
  </joint>
  <link name="lf_upper_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.0245 0.034"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00374 -0.0223 -0.0327"/>
      <mass value="1.152"/>
      <inertia ixx="0.00584" ixy="8.72e-05" ixz="-0.000289" iyy="0.0058" iyz="0.000808" izz="0.00103"/>
    </inertial>
  </link>
  <joint name="lf_lower_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="lf_upper_leg_link"/>
    <child link="lf_lower_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="35.55" lower="-2.7227" upper="-0.83776" velocity="20.06"/>
  </joint>
  <link name="lf_lower_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.016 0.016"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.00548 -0.000975 -0.115"/>
      <mass value="0.154"/>
      <inertia ixx="0.00108" ixy="3.4e-07" ixz="1.72e-05" iyy="0.0011" iyz="8.28e-06" izz="3.29e-05"/>
    </inertial>
  </link>
  <joint name="lf_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="lf_lower_leg_link"/>
    <child link="lf_foot_link"/>
  </joint>
  <link name="lf_foot_link">
        <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.06"/>
      <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
    </inertial>
  </link>
  <joint name="rh_hip_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.1934 -0.0465 0"/>
    <parent link="trunk"/>
    <child link="rh_hip_link"/>
    <axis xyz="1 0 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.0472" upper="1.0472" velocity="30.1"/>
  </joint>
  <link name="rh_hip_link">
        <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 -0.0955 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.046"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0054 -0.00194 -0.000105"/>
      <mass value="0.678"/>
      <inertia ixx="0.00048" ixy="-3.01e-06" ixz="-1.11e-06" iyy="0.000884" iyz="1.42e-06" izz="0.000596"/>
    </inertial>
  </link>
  <joint name="rh_upper_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.0955 0"/>
    <parent link="rh_hip_link"/>
    <child link="rh_upper_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.5708" upper="3.4907" velocity="30.1"/>
  </joint>
  <link name="rh_upper_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.0245 0.034"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00374 0.0223 -0.0327"/>
      <mass value="1.152"/>
      <inertia ixx="0.00584" ixy="-8.72e-05" ixz="-0.000289" iyy="0.0058" iyz="-0.000808" izz="0.00103"/>
    </inertial>
  </link>
  <joint name="rh_lower_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="rh_upper_leg_link"/>
    <child link="rh_lower_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="35.55" lower="-2.7227" upper="-0.83776" velocity="20.06"/>
  </joint>
  <link name="rh_lower_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.016 0.016"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.00548 -0.000975 -0.115"/>
      <mass value="0.154"/>
      <inertia ixx="0.00108" ixy="3.4e-07" ixz="1.72e-05" iyy="0.0011" iyz="8.28e-06" izz="3.29e-05"/>
    </inertial>
  </link>
  <joint name="rh_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="rh_lower_leg_link"/>
    <child link="rh_foot_link"/>
  </joint>
  <link name="rh_foot_link">
        <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.06"/>
      <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
    </inertial>
  </link>
  <joint name="lh_hip_joint" type="revolute">
    <origin rpy="0 0 0" xyz="-0.1934 0.0465 0"/>
    <parent link="trunk"/>
    <child link="lh_hip_link"/>
    <axis xyz="1 0 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.0472" upper="1.0472" velocity="30.1"/>
  </joint>
  <link name="lh_hip_link">
        <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0.0955 0"/>
      <geometry>
        <cylinder length="0.04" radius="0.046"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0054 0.00194 -0.000105"/>
      <mass value="0.678"/>
      <inertia ixx="0.00048" ixy="3.01e-06" ixz="-1.11e-06" iyy="0.000884" iyz="-1.42e-06" izz="0.000596"/>
    </inertial>
  </link>
  <joint name="lh_upper_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.0955 0"/>
    <parent link="lh_hip_link"/>
    <child link="lh_upper_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="23.7" lower="-1.5708" upper="3.4907" velocity="30.1"/>
  </joint>
  <link name="lh_upper_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.0245 0.034"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00374 -0.0223 -0.0327"/>
      <mass value="1.152"/>
      <inertia ixx="0.00584" ixy="8.72e-05" ixz="-0.000289" iyy="0.0058" iyz="0.000808" izz="0.00103"/>
    </inertial>
  </link>
  <joint name="lh_lower_leg_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="lh_upper_leg_link"/>
    <child link="lh_lower_leg_link"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.01" friction="0.2"/>
    <limit effort="35.55" lower="-2.7227" upper="-0.83776" velocity="20.06"/>
  </joint>
  <link name="lh_lower_leg_link">
        <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1065"/>
      <geometry>
        <box size="0.213 0.016 0.016"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.00548 -0.000975 -0.115"/>
      <mass value="0.154"/>
      <inertia ixx="0.00108" ixy="3.4e-07" ixz="1.72e-05" iyy="0.0011" iyz="8.28e-06" izz="3.29e-05"/>
    </inertial>
  </link>
  <joint name="lh_foot_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.213"/>
    <parent link="lh_lower_leg_link"/>
    <child link="lh_foot_link"/>
  </joint>
  <link name="lh_foot_link">
        <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.06"/>
      <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
    </inertial>
  </link>
</robot>
